backend/app/testbot/llm.py

```
@@ -2,69 +2,428 @@
+    
+    reduced_prompt = []
+    for section, lines in sections.items():
+        if section == "preamble":
+            reduced_prompt.extend(lines)
+        else:
+            reduced_prompt.append(f"## {section}")
+            reduced_prompt.extend(lines)
+    
+    return "\n".join(reduced_prompt)
+
+def process_query_llm(chatbot_id: str, query: str, context_docs: list, conversation_history=None) -> str:
```

Ruff (0.8.2)

Missing type annotation for function argument `conversation_history`

(ANN001)